pkg/storaged/stratis-details.jsx

```diff
@@ -263,6 +209,74 @@ export const StratisPoolDetails = ({ client, pool }) => {
         });
     }
 
+    function add_tang() {
+        return with_keydesc(client, pool, (keydesc, keydesc_set) => {
+            dialog_open({
+                Title: _("Add Tang keyserver"),
+                Fields: [
+                    TextInput("tang_url", _("Keyserver address"),
+                              {
+                                  validate: validate_url
+                              }),
+                    Skip("medskip",
+                         {
+                             visible: () => !keydesc_set
+                         }),
+                    PassInput("passphrase", _("Disk passphrase"),
+                              {
+                                  visible: () => !keydesc_set,
+                                  validate: val => !val.length && _("Passphrase cannot be empty"),
+                                  explanation: _("Adding a keyserver requires unlocking the pool. Please provide the existing pool passphrase.")
+                              })
+                ],
+                Action: {
+                    Title: _("Add"),
+                    action: function (vals, progress) {
+                        return get_tang_adv(vals.tang_url)
+                                .then(adv => confirm_tang_trust(vals.tang_url, adv, keydesc, vals.passphrase));
+                    }
+                }
+            });
+        });
+    }
+
+    function confirm_tang_trust(url, adv, keydesc, passphrase) {
+        dialog_open({
+            Title: _("Verify key"),
+            Body: <TangKeyVerification url={url} adv={adv} />,
+            Action: {
+                Title: _("Trust key"),
+                action: function (vals, progress) {
+                    function bind() {
+                        return pool.BindClevis("tang", JSON.stringify({ url, adv })).then(std_reply);
+                    }
+
+                    if (keydesc)
+                        return with_stored_passphrase(client, keydesc, passphrase, bind);
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18818-20230524-094442-1b89e402-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)